WalletWasabi/CoinJoin/Client/Clients/CoinJoinClient.cs

```diff
@@ -445,7 +445,12 @@ private async Task RegisterOutputAsync(ClientRound ongoingRound)
 						else
 						{
 							// Should never happen, but oh well we can autocorrect it so why not.
-							ExposedLinks[input] = ExposedLinks[input].Append(new HdPubKeyBlindedPair(DestinationKeyManager.GetKeyForScriptPubKey(activeOutput.Address.ScriptPubKey) ?? KeyManager.GetKeyForScriptPubKey(activeOutput.Address.ScriptPubKey), false));
+							if (!DestinationKeyManager.TryGetKeyForScriptPubKey(activeOutput.Address.ScriptPubKey, out HdPubKey? hdPubKey)
+								&& !KeyManager.TryGetKeyForScriptPubKey(activeOutput.Address.ScriptPubKey, out hdPubKey))
+							{
+								throw new NotSupportedException("Couldn't get the key for the script.");
```

Comment:
This is much better. I would like to ask you to log the address, otherwise the info in the log will not be so useful.

Reply:
Done